cmake/EthCompilerSettings.cmake

```diff
@@ -197,6 +197,8 @@ if (SANITIZE)
 	# -fsanitize-address-use-after-scope throws an error when a variable is used beyond its scope
 	if (san STREQUAL "address")
 		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-omit-frame-pointer -fsanitize=address -fsanitize-address-use-after-scope")
+	elseif (san STREQUAL "undefined")
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unsigned-integer-overflow,unreachable,vla-bound,vptr -fno-sanitize-recover=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unreachable,vla-bound,vptr")
```

Comment:
Good observation. I don't think we are relying on this since the fuzzer would have alerted me otherwise, and I have not received such a bug report so far.

Actually, the motivation for this PR are null dereferences on shared_ptr that were reported by the fuzzer on a very regular basis.

Reply:
ok, sounds reasonable. if it does not find any false-positives now then I guess we put off worrying about it until we actually run into them. We can disable this particular sanitizer then.